src/components/data.jsx

```diff
@@ -116,45 +123,51 @@ export default function DataLoader (){
     return function () {counter += 1; return counter}
   })();//closure with self invoked function:)
 
-
-
+  const genid = idMaker()
+  
+  
+    
   return (
     
+      
       <div>
-        
-        <p>{displayresults()}</p>
-        <p>{error?`Error: ${error}`:null}</p>
+         
+        <p></p>
+        <p>{error?`Error: ${error}`:displayresults()}</p>
 
           <ul>
-            
-            {data?data.map((item,index) =>
+
+            {data && (ref.current!==data)?data.map((item,index) =>
             <>
             <div id='repo-items'>
               <div id='repo-list-items'>
                 <div id='order-number'>
+
                 {ItemIncrement()}
+
                 </div>
-                {/*console.log(keyIndex())*/}
+
+                {console.log(genid.next().value)}
+
                 <li id='repo-list-items-name' 
-                key = {keyIndex()+'id'+item.id}>{item.name}</li>
-                {/*console.log(keyIndex())*/}
+                key = {genid.next().value+'id' + item.id}>{item.name}</li>
```

Comment:
you don't need to generate an index id
just use the index id coming from the map as argument
then prefix it with a namespace like:
```
key={`repo-items-blah-blah-${index}`}
```